lib/sidekiq_unique_jobs/middleware/server/unique_jobs.rb

```diff
@@ -15,8 +16,11 @@ def call(worker, item, _queue, redis_pool = nil)
           unlocked = before_yield? ? unlock(lock_key).inspect : 0
 
           yield
+        rescue Sidekiq::Shutdown
+          shutdown = true
+          raise
         ensure
-          if after_yield? || !defined? unlocked || unlocked != 1
+          if !shutdown && (after_yield? || !defined? unlocked || unlocked != 1)
```

Comment:
What happens when shutdown is true and after_yield? is also true? shouldn't we unlock in that case?